superset/dataframe.py

```diff
@@ -14,257 +14,21 @@
-        return dtype.name in ("object", "bool")
-
-    @classmethod
-    def is_id(cls, column_name):
-        return column_name.startswith("id") or column_name.endswith("id")
-
-    @classmethod
-    def agg_func(cls, dtype, column_name):
-        # consider checking for key substring too.
-        if cls.is_id(column_name):
-            return "count_distinct"
-        if (
-            hasattr(dtype, "type")
-            and issubclass(dtype.type, np.generic)
-            and is_numeric(dtype)
-        ):
-            return "sum"
-        return None
-
-    @property
-    def columns(self):
-        """Provides metadata about columns for data visualization.
-
-        :return: dict, with the fields name, type, is_date, is_dim and agg.
-        """
-        if self.df.empty:
-            return None
-
-        columns = []
-        sample_size = min(INFER_COL_TYPES_SAMPLE_SIZE, len(self.df.index))
-        sample = self.df
-        if sample_size:
-            sample = self.df.sample(sample_size)
-        for col in self.df.dtypes.keys():
-            db_type_str = self._type_dict.get(col) or self.db_type(self.df.dtypes[col])
-            column = {
-                "name": col,
-                "agg": self.agg_func(self.df.dtypes[col], col),
-                "type": db_type_str,
-                "is_date": self.is_date(self.df.dtypes[col], db_type_str),
-                "is_dim": self.is_dimension(self.df.dtypes[col], col),
-            }
-
-            if not db_type_str or db_type_str.upper() == "OBJECT":
-                v = sample[col].iloc[0] if not sample[col].empty else None
-                if isinstance(v, str):
-                    column["type"] = "STRING"
-                elif isinstance(v, int):
-                    column["type"] = "INT"
-                elif isinstance(v, float):
-                    column["type"] = "FLOAT"
-                elif isinstance(v, (datetime, date)):
-                    column["type"] = "DATETIME"
-                    column["is_date"] = True
-                    column["is_dim"] = False
-                # check if encoded datetime
-                if (
-                    column["type"] == "STRING"
-                    and self.datetime_conversion_rate(sample[col])
-                    > INFER_COL_TYPES_THRESHOLD
-                ):
-                    column.update({"is_date": True, "is_dim": False, "agg": None})
-            # 'agg' is optional attribute
-            if not column["agg"]:
-                column.pop("agg", None)
-            columns.append(column)
-        return columns
+def df_to_dict(dframe: pd.DataFrame) -> Dict:
+    data = dframe.to_dict(orient="records")
+    # TODO: refactor this
+    for d in data:
+        for k, v in list(d.items()):
+            # if an int is too big for Java Script to handle
+            # convert it to a string
+            if isinstance(v, int):
+                if abs(v) > JS_MAX_INTEGER:
+                    d[k] = str(v)
+    return data
```

Comment:
Would it be preferable or simpler to convert all numeric types in the dataframe to strings before dict-ifying it to (I presume) send to the client?

Reply:
Yes, we could do that, but I'm unclear on the frontend implications of shipping Ints as strings. When the column contains nullable integer values, the conversion from PyArrow to Pandas DataFrame is done via an `object` dtype (string).

Reply:
It seems dangerous either way. Could JS do arrow and magically do the right thing around bigints? Didn't read, but pasting https://arrow.apache.org/docs/js/

Reply:
We may be able to send Arrow or Msgpack-encoded data over the wire to the client in future work. In the interest of limiting scope here, I propose we stick with the existing approach for handling BigInts for now, and revisit in a fast-follow that's more focused on performance optimization.